pythclient/price_feeds.py

```diff
@@ -502,28 +581,242 @@ def extract_price_info_from_accumulator_update(
+                trailing_header_size=combined_data.trailing_header_size,
+                update_type=combined_data.update_type,
+                vaa_length=combined_data.vaa_length,
+                vaa=combined_data.vaa,
+                num_updates=len(chunk),
+                updates=chunk,
+            )
+            chunked_updates.append(chunked_update)
+
+    # Serialize it into the accumulator update format
+    serialized_data_list = []
+    for update in chunked_updates:
+        serialized_data = serialize_accumulator_update(update, encoding)
+        serialized_data_list.append(serialized_data)
+
+    return serialized_data_list
+
+
+def serialize_accumulator_update(data, encoding):
+    """
+    This function serializes an accumulator update data into a string.
+
+    Parameters:
+    data (dict): The accumulator update data to serialize. The dictionary should include the following keys:
+        - "magic": The magic of the update data.
+        - "major_version": The major version of the update data.
+        - "minor_version": The minor version of the update data.
+        - "trailing_header_size": The size of the trailing header in the update data.
+        - "update_type": The type of the update.
+        - "vaa_length": The length of the VAA.
+        - "vaa": The VAA itself.
+        - "num_updates": The number of updates in the update data.
+        - "updates": A list of dictionaries, where each dictionary represents an update and includes the following keys:
+            - "message_size": The size of the message in the update.
+            - "message": The message itself.
+            - "proof_size": The size of the proof in the update.
+            - "proof": The proof itself.
+    encoding (str): The encoding of the serialized data. Can be either "hex" or "base64".
+
+    Returns:
+    str: The serialized accumulator update data as a string. If the encoding is "hex", the function returns a hexadecimal string. If the encoding is "base64", the function returns a base64 string.
+    """
+    serialized_data = bytearray()
+    serialized_data.extend(data.magic)
+    serialized_data.append(data.major_version)
+    serialized_data.append(data.minor_version)
+    serialized_data.append(data.trailing_header_size)
+    serialized_data.append(data.update_type)
+    serialized_data.extend(data.vaa_length.to_bytes(2, byteorder="big"))
+    serialized_data.extend(data.vaa)
+    serialized_data.append(data.num_updates)
+    for update in data.updates:
+        serialized_data.extend(update.message_size.to_bytes(2, byteorder="big"))
+        serialized_data.extend(update.message)
+        serialized_data.append(update.proof_size)
+        for proof in update.proof:
+            serialized_data.extend(proof)
+    if encoding == "hex":
+        return serialized_data.hex()
+    else:
+        return base64.b64encode(serialized_data).decode("ascii")
+
+
+def parse_accumulator_update(update_data, encoding):
```

Comment:
this is quite similar to extract_price_infos_from_accumulator... Can't we remove the duplicate code? the former method can use the data on this method to get the price infos

Reply:
you're right, updated ser